kafka/client.py

```diff
@@ -258,15 +258,19 @@ def reset_all_metadata(self):
         self.topic_partitions.clear()
 
     def has_metadata_for_topic(self, topic):
```

Comment:
I did it this way as a premature optimization because I saw has_metadata_for_topic and get_partition_ids_for_topic were on the callpath of send_messages which is probably the most used method and always calls these metadata methods with a byte string. But on closer inspection, it should only call them once on initialization, so its probably clearer to use a consistent pattern.